src/lib/crypto/kimchi_backend/gadgets/keccak.ml

```diff
@@ -0,0 +1,720 @@
+open Core_kernel
+module Bignum_bigint = Snarky_backendless.Backend_extended.Bignum_bigint
+module Snark_intf = Snarky_backendless.Snark_intf
+
+open Kimchi_backend_common.Plonk_constraint_system.Plonk_constraint
+
+let tests_enabled = true
+
+(* DEFINITIONS OF CONSTANTS FOR KECCAK *)
+
+(* Length of the square matrix side of Keccak states *)
+let keccak_dim = 5
+
+(* value `l` in Keccak, ranges from 0 to 6 (7 possible values) *)
+let keccak_ell = 6
+
+(* width of the lane of the state, meaning the length of each word in bits (64) *)
+let keccak_word = Int.pow 2 keccak_ell
+
+(* length of the state in bits, meaning the 5x5 matrix of words in bits (1600) *)
+let keccak_state_length = Int.pow keccak_dim 2 * keccak_word
+
+(* number of rounds of the Keccak permutation function depending on the value `l` (24) *)
+(*let keccak_rounds = 12 + (2 * keccak_ell)*)
```

Comment:
Oh sure, that was commented because at first I was manually changing the number of rounds to be 1 or 2, but now there is no need to have it like that anymore.